src/app/players/players.component.ts

```diff
@@ -98,30 +104,30 @@ export class PlayersComponent implements OnInit, OnDestroy {
 
         const gradeSubscription = this.playersFormGroup.controls[
           `${player.name}-grade`
-        ].valueChanges
-          .distinctUntilChanged()
-          .debounceTime(200)
+        ].valueChanges.pipe(
+          distinctUntilChanged(),
+          debounceTime(200),
           // avoid triggering the change user grade if
           // the grade is empty as it'd parse the grade
           // as integer and set an annoying 0 everytime
           // we clean all the input before typing something
           // else (mostly annoying on mobile)
-          .filter(grade => grade)
-          .do(grade =>
-            this.playersService.changeUserGrade(player.name, +grade)
-          );
+          filter(grade => grade),
```

Comment:
Could you write that one as 

```
filter(grade => !!grade)
```

that'll be clearer

Reply:
Sure. I can do that.